mesonbuild/interpreterbase/decorators.py

```diff
@@ -323,7 +323,7 @@ def __init__(self, name: str, types: T.Union[T.Type[_T], T.Tuple[T.Type[_T], ...
                  deprecated: T.Optional[str] = None,
                  deprecated_values: T.Optional[T.Dict[str, str]] = None,
                  validator: T.Optional[T.Callable[[_T], T.Optional[str]]] = None,
-                 convertor: T.Optional[T.Callable[[_T], TYPE_var]] = None,
+                 convertor: T.Optional[T.Callable[[_T], object]] = None,
```

Comment:
Is there a specific reason for why this is required or is this is a debugging leftover?

Reply:
No, it's needed. The convertor function is in no way bound to return a TYPE_var, in fact of the three uses 2 don't return a TYPE_var. One of them converts the `bool` of the `native` keyword argument into a `MachineChoice`, the second converts the `install_mode` keyword argument into a `FileMode` object.

Reply:
Ok, fair enough.